### ISSUE TITLE
Traders' stock/demand values don't get updated on sell

### ISSUE THREAD

**Issue by olegst90** [pioneerspacesim/pioneer#5889]
When I trade with traders from bulletin board, their "in stock" and "demand" values don't get updated when I buy or sell items. This happens both with fuel and illegal traders. As a result, I can sell more than their demand is, as opposed to the normal trade where you can't sell more than they need. Using 20240710 version for Windows

**Comment by impaktor** [comment]
Yes. I discussed it recently with @Web-eWorks, basically we know about it, but good to have it on the tracker.

Ideally, I'd rather see the demand-functionallity just lower the price once demand is met, rather than be a hard stop, where you can't get rid of your cargo no matter what price you're willing to put it on the market.
